src/diff/children.js

```diff
@@ -53,6 +53,13 @@ export function diffChildren(
 	isHydrating,
 	refQueue
 ) {
+	if (
+		parentDom instanceof DocumentFragment &&
+		newParentVNode.type === 'template' &&
+		excessDomChildren?.length === 0
+	) {
+		excessDomChildren = Array.from(parentDom.childNodes);
+	}
```

Comment:
We'd want to move the cheaper tag type check first, `instanceof` is considerably more expensive, especially as a first check in a chain when most will bail (`<template>` is pretty uncommonly used).

```suggestion
	if (
		newParentVNode.type === 'template' &&
		excessDomChildren?.length === 0 &&
		parentDom instanceof DocumentFragment
	) {
		excessDomChildren = slice.call(parentDom.childNodes);
	}
```

Will also need to add an import for `slice`.